### PR TITLE
UI form best practices for hiding inputs - research paper

### PR DESCRIPTION
**Pull request by catenglaender**
In this paper, we investigate how optional or advanced sections in complex ILIAS forms could be presented and hidden if they are not currently relevant.

The goal is to improve the user experience by
* reducing visual noise that can be overwhelming and intimidating
* speeding up the process of filling out complex forms

To read the paper [click here!](https://github.com/ILIAS-eLearning/ILIAS/blob/8d3b09132863f585be18f5ba3669eb070aa007b9/src/UI/docu/UI-forms_best-practices-for-hiding-inputs-in-complex-forms.md)

The images will find their permanent home on the ILIAS file server.

This research was commissioned by the University of Bern and carried out by CaT Concepts and Training GmbH.
There is a presentation "The perfect form" summarizing this paper during the March 2024 ILIAS DevConf.

### CODE REVIEW

File: src/UI/docu/UI-forms_best-practices-for-hiding-inputs-in-complex-forms.md [hunk 1]
```diff
@@ -0,0 +1,600 @@
...
+
+## Object-Oriented Branching
+
+As use cases for forms get more complex, it gets more and more important to think about the context and sequence that the individual forms are in and how everything could be connected in an intuitive way.
+
+While the previous two chapters suggested occasionally listing information as editable items in a view mode, the following approach is built all around object-oriented structures and summary or list sections exposing deeper layers to the parent.
+
+A medium article by Francis Wu[^francis-wu_complex-forms-OOUX] suggests branching sub-objects into sub-forms:
+
+![](./img/Francis-Wu_dashboard-sub-forms.png)
```
Comment (thibsy) [inline src/UI/docu/UI-forms_best-practices-for-hiding-inputs-in-complex-forms.md]: Something went wrong with embedding the screenshot here.

**Review by thibsy — COMMENTED** [review]
Hi @catenglaender,

Thx a lot for this very detailed analysis on hiding inputs in complex forms! Looking forward to applying these concepts to some concrete use cases.

Kind regards,
@thibsy

**Comment by yvseiler** [comment]
Thank you for the detailed and helpfully structured analysis with references to existing patterns about forms. I find the conclusion that we will probably need several types, but limit them for precisely defined purposes (e.g. creation vs. viewing and checking) very reasonable. An exciting field of work to delve into.

**Comment by Amstutz** [comment]
hi @catenglaender , is it possible to remove the draft status here? IMO this could be merged and so referenced later. I am aware that "The images will find their permanent home on the ILIAS file server" and can so be linked, however, maybe just put that point on the roadmap.

File: src/UI/docu/UI-forms_best-practices-for-hiding-inputs-in-complex-forms.md (hunk 1 above)
Comment (catenglaender) [reply src/UI/docu/UI-forms_best-practices-for-hiding-inputs-in-complex-forms.md]: Well spotted! Thank you! The image should now be linked correctly.

**Comment by catenglaender** [comment]
Thank you all for your input not only here, but also at the DevConf :slightly_smiling_face: 

Images are now loaded from files.ilias.de (thx @matthiaskunkel) :heavy_check_mark: 

Paper is ready to be submitted.

**Comment by matthiaskunkel** [comment]
Jour Fixe, 15 APR 2024: We highly appreciate this paper and accept it for the documentation of the UI framework. @ All: please have a look at the paper if you are working on complex input UIs.

**Comment by Amstutz** [comment]
Thx a lot!

**Pull request merged**